gamd/parser.py

```diff
@@ -185,11 +192,56 @@ def parse_charmm_tag(input_files_tag):
         elif charmm_tag.tag == "coordinates":
             charmm_config.coordinates = assign_tag(charmm_tag, str)
+            if "type" in charmm_tag.attrib:
+                type_attrib = charmm_tag.attrib["type"]
+                charmm_config.coordinates_filetype = type_attrib
+        elif charmm_tag.tag == "box-vectors":
+            box_dict = {}
+            for box_info in charmm_tag:
+                if box_info.tag in ["a", "b", "c"]:
+                    box_dict[box_info.tag] = (assign_tag(box_info, float, 
+                                                        useunit=unit.nanometer))
+                elif box_info.tag in ["alpha", "beta", "gamma"]:
+                    box_dict[box_info.tag] = (assign_tag(box_info, float, 
+                                                        useunit=unit.degree))
+                else:
+                    raise Exception("Unkown parameter '" +box_info.tag+ "'. "\
+                         "Accepted box-vector parameters are 'a', 'b', 'c', "\
+                         "'alpha', 'beta' and 'gamma'." )
+
+            # if angles are not set, use default values
+            if "alpha" not in box_dict.keys():
+                box_dict["alpha"] = charmm_config.alpha
+            if "beta" not in box_dict.keys():
+                box_dict["beta"] = charmm_config.beta
+            if "gamma" not in box_dict.keys():
+                box_dict["gamma"] = charmm_config.beta
+            # raise Exception if any box vector is not defined
+            for boxlength in ["a", "b","c"]:
+                if boxlength not in box_dict.keys():
+                    raise Exception("box-vector 'a', 'b' and 'c' must be "\
+                    "defined! Missing parameter " + "'" + boxlength + "'.")
+
+            charmm_config.box_vectors = box_dict
+
         elif charmm_tag.tag == "parameters":
             charmm_config.parameters = []
             for xml_params_filename in charmm_tag:
-                charmm_config.parameters.append(
-                    assign_tag(xml_params_filename, str))
+                if "type" in xml_params_filename.attrib:
+                    if xml_params_filename.attrib["type"] == "charmm-gui-toppar":
+                        # parsing list of parameter files like CHARMM-GUI does
+                        extlist = ['rtf', 'prm', 'str']
+                        parFiles = ()
+                        for line in open(xml_params_filename.text, 'r'):
+                            if '!' in line: line = line.split('!')[0]
+                            parfile = line.strip()
+                            if len(parfile) != 0:
+                                ext = parfile.lower().split('.')[-1]
+                                if not ext in extlist: continue
+                                charmm_config.parameters.append(parfile)
+                else:
+                    charmm_config.parameters.append(
+                        assign_tag(xml_params_filename, str))
```

Comment:
Can you clarify a bit more? I could create a function on parser.py or move this to config.py.

Reply:
Sure.  I was thinking just add another method.  Something like the following:
```
def parse_and_assign_charmm_gui_toppar_file(self, charmm_config, xml_params_filename):
    extlist = ['rtf', 'prm', 'str']
    parFiles = ()
    for line in open(xml_params_filename.text, 'r'):
        if '!' in line: line = line.split('!')[0]
        parfile = line.strip()
        if len(parfile) != 0:
            ext = parfile.lower().split('.')[-1]
            if not ext in extlist: continue
            charmm_config.parameters.append(parfile)
    return charmm_config
```
Then on line 232, just replace the code with
```
charmm_config = self.parse_and_assign_charmm_gui_toppar_file(charmm_config, xml_params_filename)
```
The thought being to try and have the methods be smaller and limited to doing a single thing as much as possible, so that they don't get to large.

Reply:
Thank you. I thought it was better to make that parsing function a method inside CharmmConfig class (config.py).

So on line 233 I am just calling:
```
charmm_config.parse_charmm_gui_toppar(xml_params_filename)
```
Which calls the method  (config.py, line 147):
```
def parse_charmm_gui_toppar(self, xml_params_filename):
        extlist = ['rtf', 'prm', 'str']
        # inputfile is the toppar.str with the list of parameter files
        for line in open(xml_params_filename.text, 'r'):
            if '!' in line: line = line.split('!')[0]
            parfile = line.strip()
            if len(parfile) != 0:
                ext = parfile.lower().split('.')[-1]
                if not ext in extlist: continue
                # Appending each file listed in inputfile to the existing 
                # list of parameters files to be read
                self.parameters.append(parfile)
        return
```
I wanted to make sure users can use both parsing methods in the same input (directly pointing to parameter files or listing them in a toppar.str file). I think having all methods in config.py is probably cleaner.